main/pipeline/pipeline.cc

```diff
@@ -288,26 +265,72 @@ parser::ParseResult runPrismParser(core::GlobalState &gs, core::FileRef file, co
         core::UnfreezeNameTable nameTableAccess(gs); // enters strings from source code as names
         // The RBS rewriter produces plain Whitequark nodes and not `NodeWithExpr` which causes errors in
         // `PrismDesugar.cc`. For now, disable all direct translation, and fallback to `Desugar.cc`.
-        auto directlyTranslate = !gs.cacheSensitiveOptions.rbsEnabled;
-        parseResult = parser::Prism::Parser::run(ctx, directlyTranslate);
-    }
+        auto source = file.data(ctx).source();
+        parser::Prism::Parser parser{source};
+        bool collectComments = gs.cacheSensitiveOptions.rbsEnabled;
+        parser::Prism::ParseResult prismResult = parser.parseWithoutTranslation(collectComments);
 
-    if (print.ParseTree.enabled) {
-        print.ParseTree.fmt("{}\n", parseResult.tree->toStringWithTabs(gs, 0));
-    }
-    if (print.ParseTreeJson.enabled) {
-        print.ParseTreeJson.fmt("{}\n", parseResult.tree->toJSON(gs, 0));
-    }
-    if (print.ParseTreeJsonWithLocs.enabled) {
-        print.ParseTreeJson.fmt("{}\n", parseResult.tree->toJSONWithLocs(gs, file, 0));
+        if (opts.stopAfterPhase == options::Phase::PARSER) {
+            return parser::ParseResult{nullptr, prismResult.getCommentLocations()};
+        }
```

Comment:
Are you relying on this stopAfter behavior?

It seems like if anything, we would want `--stop-after=parser` with RBS comment parsing enabled to actually stop after RBS comment parsing, which would mean deleting this.

In any case, I would like to hear more about this choice.